base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -23,23 +23,43 @@
 use crate::{
     base_node::comms_interface::{error::CommsInterfaceError, NodeCommsRequest, NodeCommsResponse},
     blocks::{blockheader::BlockHeader, Block},
-    chain_storage::{async_db, BlockchainBackend, BlockchainDatabase, HistoricalBlock},
+    chain_storage::{
+        async_db,
+        BlockAddResult,
+        BlockchainBackend,
+        BlockchainDatabase,
+        ChainStorageError,
+        HistoricalBlock,
+    },
 };
+use futures::SinkExt;
+use tari_broadcast_channel::Publisher;
 use tari_transactions::transaction::{TransactionKernel, TransactionOutput};
 
+/// Events that can be published on the Validated Block Event Stream
+#[derive(Debug)]
+pub enum BlockEvent {
+    Verified((Block, BlockAddResult)),
+    Invalid((Block, ChainStorageError)),
+}
```

Comment:
I'll merge this in as in for now, but I think a client listening for block events isn't really interested in `BlockAddResult`; and there are other reasons why a block might be rejected other than a chain storage error.

We can update as appropriate together with  the block validation pipeline (mentioned below).